sdk/cosmosdb/cosmos/src/ClientContext.ts

```diff
@@ -116,6 +117,7 @@ export class ClientContext {
       return response;
     } catch (err: any) {
       this.captureSessionToken(err, path, OperationType.Upsert, (err as ErrorResponse).headers);
+      setDiagnostics(`${err}`);
```

Comment:
Since this is not called on any object instance, rather called directly, I am curious to know how do we make sure that it doesn't overrides other diagnostics information which are already stored inside `_diagnosticHeader` in `Diagnostics.ts` class.